cobigen/cobigen-javaplugin-parent/cobigen-javaplugin/src/main/java/com/devonfw/cobigen/javaplugin/merger/libextension/ModifyableModelBuilder.java

```diff
@@ -348,17 +390,29 @@ public void endMethod(MethodDef def) {
         currentMethod.setSourceCode(def.getBody());
     }
 
-    private <G extends JavaGenericDeclaration> DefaultJavaTypeVariable<G> createTypeVariable(
-        TypeVariableDef typeVariableDef, G genericDeclaration) {
+    /** Logger instance. */
+    private static final Logger LOG = LoggerFactory.getLogger(ModifyableModelBuilder.class);
 
+    private <G extends JavaGenericDeclaration> ModifyableJavaTypeVariable<G> createTypeVariable(
+        TypeVariableDef typeVariableDef, G genericDeclaration) {
         if (typeVariableDef == null) {
             return null;
         }
-        DefaultJavaTypeVariable<G> result =
-            new DefaultJavaTypeVariable<>(typeVariableDef.getName(), genericDeclaration);
+
+        JavaClass declaringClass = getContext(genericDeclaration);
+
+        if (declaringClass == null) {
+            return null;
+        }
+
+        TypeResolver typeResolver =
+            TypeResolver.byClassName(declaringClass.getBinaryName(), classLibrary, source.getImports());
+
+        ModifyableJavaTypeVariable<G> result =
+            new ModifyableJavaTypeVariable<G>(typeVariableDef.getName(), typeResolver);
 
         if (typeVariableDef.getBounds() != null && !typeVariableDef.getBounds().isEmpty()) {
-            List<JavaType> bounds = new LinkedList<>();
+            List<JavaType> bounds = new LinkedList<JavaType>();
```

Comment:
the type in the constructor is not needed, it even throws a warning, that it is redundant. You simply can use <>.